src/SFA.DAS.SharedOuterApi/Infrastructure/RestApiClient.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebUtilities;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.SharedOuterApi.Infrastructure
+{
+    public class RestApiClient : IRestApiClient
+    {
+        private readonly HttpClient _httpClient;
+
+        public RestApiClient(HttpClient httpClient)
+        {
+            _httpClient = httpClient;
+        }
+
+        public async Task<T> Get<T>(Uri uri, object queryData = null, CancellationToken cancellationToken = default)
+        {
+            var response = await GetResponse(uri, queryData, cancellationToken).ConfigureAwait(false);
+            return await response.Content.ReadFromJsonAsync<T>(null, cancellationToken).ConfigureAwait(false);
+        }
+
+        public Task<T> Get<T>(string uri, object queryData = null, CancellationToken cancellationToken = default)
```

Comment:
Can we not just settle on one method for each of these operations, seems un-necessary to have four methods for `Get`

Reply:
I wish we could, but I do have a compromise suggestion.

I have removed the endpoint which were using the URI type and have settled with the uri as a string

however I still end up with 4 Get functions. The logic for this is if we just stick to using this
 
Get(string uri, object queryData = null, CancellationToken cancellationToken = default)

it is really easy to make the following call and the code will compile and appear to work

Get("/test/", cancellationToken)

however we have just passed the cancellation Token in as the objectData and this will be added to the get path. Therefore to mitigate this it's worth having 

Get(string uri, CancellationToken cancellationToken = default)
Get(string uri, object queryData, CancellationToken cancellationToken = default)

And we also need a typed Get as the 3.1 json serializer only works with classes